src/rpc/server.cpp

```diff
@@ -314,7 +314,7 @@ void SetRPCWarmupStatus(const std::string& newStatus)
 void SetRPCWarmupFinished()
 {
     LOCK(cs_rpcWarmup);
-    assert(fRPCInWarmup);
+    CHECK_NONFATAL(fRPCInWarmup);
```

Comment:
I don't think you should change this one, it's used as part of initialization not RPC.

Reply:
Updated.

Reply:
Hmm. Looks like the linter is working (a little too well). Any suggestions on how I might relax the linter requirements? cc @MarcoFalke?

Reply:
My suggestion would be to add an `ASSERT` macro to `check.h` that just does `assert()` for now.

In the short term this helps to fix the linter. In the long term we can get rid of the default `assert` and replace it with our own shiny Bitcoin Core `ASSERT` (doing fancy things like logging or writing the traceback to stderr, ...)

Reply:
or exclude `rpcserver.cpp`, it contains mostly meta-code

Reply:
These needs to be fixed? :)

```
CHECK_NONFATAL(condition) should be used instead of assert for RPC code.
src/rpc/server.cpp:317:    assert(fRPCInWarmup);
src/wallet/rpcdump.cpp:165:        assert(key.VerifyPubKey(pubkey));
src/wallet/rpcdump.cpp:645:            assert(key.VerifyPubKey(pubkey));
src/wallet/rpcdump.cpp:918:        assert(script_ctx == ScriptContext::TOP);
^---- failure generated from test/lint/lint-assertions.sh
```